nsv13/code/modules/overmap/weapons.dm

```diff
@@ -79,22 +74,21 @@
 		last_target = null
 	var/found_target = FALSE //Have we found a torpedo to shoot down? If we can't find a torpedo to shoot, look for enemy ships in range.
 	if(torpedoes_to_target.len)  //Are there any torpedoes we need to worry about? Torpedoes enter this list as theyre shot (when they target us).
-		for(var/X in torpedoes_to_target) //Check through the torpedoes that our PDCs need to target
-			var/atom/target = X
+		for(var/atom/target in torpedoes_to_target) //Check through the torpedoes that our PDCs need to target
 			if(!target || QDELETED(target)) //Clear null bullets that may have runtimed
-				torpedoes_to_target -= X
+				torpedoes_to_target -= target
 				continue
 			var/target_range = get_dist(target,src)
 			if(target_range <= initial(weapon_range)) //The torpedo is in range, let's target it!
 				found_target = TRUE
 				if(prob(pdc_miss_chance)) //Gives them a chance to actually hit a torpedo, so it's not a perfect smokescreen.
-					var/turf/T = get_turf(pick(orange(6,target))) //Pick a random tile within 6 turfs, this isn't a flat out miss 100% of the time though
+					var/turf/T = get_turf(pick(orange(4,target))) //Pick a random tile within 6 turfs, this isn't a flat out miss 100% of the time though
 					fire_pdcs(T, lateral=TRUE)
 				else
 					fire_pdcs(target, lateral=TRUE)
 	if(!found_target) //Can't see a torpedo to shoot, try find an enemy ship to shoot
 		for(var/obj/structure/overmap/ship in GLOB.overmap_objects)
-			if(!istype(ship, /obj/structure/overmap))
+			if(!ship || !istype(ship, /obj/structure/overmap))
```

Comment:
null ship in overmap_objects? Also not merging the if below with this one?

Reply:
A better method to manage GLOB.overmap_objects might be separating it into factions or something. That would let you to spawn in meteors as flying objects which are neutral to avoid.
You can then more easily manage it in separate code that signals use to manage for the ships instead as they are needed.

Reply:
noted